core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -824,11 +825,15 @@ private[spark] class MapOutputTrackerWorker(conf: SparkConf) extends MapOutputTr
       endPartition: Int): Iterator[(BlockManagerId, Seq[(BlockId, Long, Int)])] = {
     logDebug(s"Fetching outputs for shuffle $shuffleId, mappers $startMapIndex-$endMapIndex" +
       s"partitions $startPartition-$endPartition")
-    val statuses = getStatuses(shuffleId, conf)
     try {
+      val statuses = getStatuses(shuffleId, conf)
       MapOutputTracker.convertMapStatuses(
         shuffleId, startPartition, endPartition, statuses, startMapIndex, endMapIndex)
     } catch {
+      case e: IOException if
+        Throwables.getCausalChain(e).asScala.exists(_.isInstanceOf[BlockNotFoundException]) =>
+        mapStatuses.clear()
+        throw new MetadataFetchFailedException(shuffleId, -1, Throwables.getStackTraceAsString(e))
```

Comment:
I'm wondering that we may could add a new type of `FetchFailedException` to indicate the whole `MapStatus` lost rather than a piece of it.

Reply:
@Ngone51 I'm both fine with add a new type of `FetchFailedException` or just use `MetadataFetchFailedException`, but I think maybe the `MetadataFetchFailedException` is already enough for expressing the lost of the `Mapstatus`(the mapId and reduceId are both set to -1)